packages/amplify-cli-core/src/help/help-helpers.ts

```diff
@@ -0,0 +1,239 @@
+import { $TSContext, $TSAny } from '../types';
+import { printer } from 'amplify-prompts';
+import chalk from 'chalk';
+
+export type CommandFlagInfo = {
+  short: string,
+  long: string,
+  flagDescription: string
+};
+
+export type SubCommandInfo = {
+  subCommand: string,
+  subCommandDescription: string,
+  subCommandUsage: string,
+  subCommandFlags: Array<CommandFlagInfo>
+};
+
+export type CommandInfo = {
+  command: string,
+  commandDescription: string,
+  commandUsage: string,
+  commandFlags: Array<CommandFlagInfo>,
+  subCommands: Array<SubCommandInfo>
+};
+
+const SPACE = ' ';
+const DEFAULT_COLUMN_SEP_SPACING = 2;
+const DEFAULT_INDENT_SPACING = 2;
+const DEFAULT_COLUMN_SEP = SPACE.repeat(DEFAULT_COLUMN_SEP_SPACING);
+const DEFAULT_INDENT = SPACE.repeat(DEFAULT_INDENT_SPACING);
+
+const USAGE = 'USAGE';
+const DESCRIPTION = 'DESCRIPTION';
+const COMMANDS = 'COMMANDS';
+const FLAGS = 'FLAGS';
+const LEARN_MORE = 'LEARN MORE';
+const SUBCOMMANDS = 'SUBCOMMANDS';
+const TAG_LINE = 'The Amplify Command Line Interface (CLI) is a unified toolchain to create, integrate, and manage the AWS cloud services for your app.';
+const DEFAULT_LINK = 'https://docs.amplify.aws/cli/';
+
+const FLAG_DELIMITER = ' | ';
+const FLAG_PREFIX_SHORT = '-';
+const FLAG_PREFIX_LONG = '--';
+
+const printHeaderText = (text: string) => printer.info(chalk.blue.bold(text));
+const printBodyText = (text: string) => printer.info(text);
+const printCategoryMessage = () => {
+  printBodyText(DEFAULT_INDENT + 'Where <category> is one of: notifications, api, auth, custom, storage,');
+  printBodyText(DEFAULT_INDENT + 'analytics, function, geo, hosting, interactions, predictions, xr');
+  printer.blankLine();
+};
+
+const AMPLIFY_CLI_DOCS_URL = `https://docs.amplify.aws/cli`;
+const HEADLESS_DOCS_LINK = `${AMPLIFY_CLI_DOCS_URL}/usage/headless`;
+const getDocsLinkForCommand = (commandName: string) =>  `${AMPLIFY_CLI_DOCS_URL}/commands/${commandName}`;
+
+export function lookUpCommand(commandsInfo: Array<CommandInfo>, commandName: string): CommandInfo | null {
+  let foundCommand: CommandInfo | null = null;
+  for (const command of commandsInfo) {
```

Comment:
[`Array.find`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) would be helpful here